src/metabase/search/scoring.clj

```diff
@@ -1,3 +1,106 @@
+;;
+;; <pre><code>
+;; (premium-features/has-feature? :official-collections)
+;;     (conj {:weight 2
+;;             :score  (official-collection-score result)
+;;             :name   "official collection score"})
+;;     (premium-features/has-feature? :content-verification)
+;;     (conj {:weight 2
+;;            :score  (verified-score result)
+;;            :name   "verified"})))
+;; </code></pre>
+;;
+;; These are easier to explain: you get points if the search result is pinned (yes or no), bookmarked (yes or no), how
+;; recently it was updated (sliding value between 1 (edited just now) and 0 (edited [180+
+;; days](https://github.com/metabase/metabase/blob/8d5f5db02c84899a053e20468986050b2034a9a4/src/metabase/search/config.clj#L29-L32)
+;; ago), how many dashboards it appears in (sliding value between 0 (zero dashboards) and 1 ([50+
+;; dashboards](https://github.com/metabase/metabase/blob/8d5f5db02c84899a053e20468986050b2034a9a4/src/metabase/search/config.clj#L34-L36))
+;; and it's type (`model-score`): the earlier a type appears in [this
+;; list](https://github.com/metabase/metabase/blob/8d5f5db02c84899a053e20468986050b2034a9a4/src/metabase/search/config.clj#L55-L58)
+;; the higher score it gets:
+;;
+;; <code> ["dashboard" "metric" "segment" "indexed-entity" "card" "dataset" "collection" "table" "action" "database"]</code>
+;;
+;; On the EE side, we also give points if something's an official collection and if it's verified.
+;;
+;; Finally, what we actually search is defined in the search
+;; config [here](https://github.com/metabase/metabase/blob/8d5f5db02c84899a053e20468986050b2034a9a4/src/metabase/search/config.clj#L73-L109),
+;; but the short answer is "the name and, if there is one, the description". We used to search raw SQL queries for
+;; cards, but that got turned off recently (but I've seen chat about turning it back on).
+;;
+;; ❦
+;;
+;; So, these 12 scorers are weighted and combined together, and the grand total affects search order. If this sounds a
+;; little complicated…it is! It also means that it can be tricky to give a proper answer about why the search ranking
+;; is "wrong", maybe you search for `monthly revenue` and are looking for a card called `monthly revenue` and are mad
+;; that a dashboard called `company stats` shows up first…but then it turns out that the dashboard's description is
+;; `Stats that everyone should be aware of, such as our order count and monthly revenue.` and the dashboard happens to
+;; be pinned, bookmarked, part of an official collection, verified, and edited a couple hours ago…whereas the card is
+;; none of those things.
+;;
+;; Also, be aware that there's [a big epic under way](https://github.com/metabase/metabase/issues/27982) to add
+;; filtering to search results, which should help people find what they're looking for (and spares us from having to
+;; make the above algorithm better).
```

Comment:
Can we phrase this paragraph in a way that won't make this out of date in a few months like "As of October 2023, there is work being done toward ... "